site/book/05-developing-functions/04-executable-configuration.md

```diff
@@ -1,128 +0,0 @@
-In this section we are going to take a look at two functions that follow the
-[executable configuration pattern] we discussed at the beginning of this
-chapter: `starlark` and `gatekeeper`.
-
-## `starlark` function
-
-[Starlark] is a python-like language designed for use in configuration files
-that has several desirable properties: deterministic evaluation, hermetic
-execution, and simplicity. The `starlark` function contains the interpreter for
-the language, and accepts a `functionConfig` of kind `StarlarkRun` where you
-provide your business logic.
-
-The following is an example package with a `starlark` function declaration:
-
-```shell
-$ kpt pkg get https://github.com/GoogleContainerTools/kpt-functions-catalog.git/examples/starlark-set-namespace@starlark/v0.2
-```
-
-It contains the following `functionConfig`:
-
-```yaml
-# set-namespace/fn-config.yaml
-apiVersion: fn.kpt.dev/v1alpha1
-kind: StarlarkRun
-metadata:
-  name: set-namespace-to-prod
-source: |
-  # set the namespace on all resources
-  def setnamespace(resources, namespace):
-    for resource in resources:
-      # mutate the resource
-      resource["metadata"]["namespace"] = namespace
-  setnamespace(ctx.resource_list["items"], "prod")
-```
-
-The `source` field includes the Starlark logic that sets the namespace on all
-resources. Go ahead and render the package:
-
-```shell
-$ kpt fn render set-namespace
-```
-
-You should now see that resources have `namespace` set to `prod`.
-
-?> Refer to the [Functions Catalog](https://catalog.kpt.dev/starlark/v0.2/) for
-details on how use this function.
-
-## `gatekeeper` function
```

Comment:
Do we want to drop the contents about gatekeeper?

Reply:
yes. It's a good and curated function which has been well documented in the catalog page. 
For SDK guide, I think the users should use the most clear and unambiguous function to get started (assuming they are beginners). But from some previous chats, it seems that there are always confusions around gatekeeper. WDYT?